src/Xrefcheck/Verify.hs

```diff
@@ -423,29 +421,37 @@ verifyReference
   -> VerifyMode
   -> IORef VerifyProgress
   -> RepoInfo
-  -> FilePath
-  -> FilePath
+  -> CanonicalPath
   -> Reference
   -> IO (VerifyResult $ WithReferenceLoc VerifyError)
 verifyReference
   config@Config{..}
   mode
   progressRef
-  (RepoInfo files dirs)
-  root
-  fileWithReference
+  repoInfo
+  file
   ref@Reference{..}
-    = retryVerification 0 $ do
-        let locType = locationType rLink
-        if shouldCheckLocType mode locType
-        then case locType of
-          FileLocalLoc      -> checkRef rAnchor fileWithReference
-          RelativeLoc       -> checkRef rAnchor
-                                (normalise $ takeDirectory fileWithReference
-                                  </> toString (canonizeLocalRef rLink))
-          AbsoluteLoc       -> checkRef rAnchor (root <> toString rLink)
-          ExternalLoc       -> checkExternalResource config rLink
-          OtherLoc          -> verifying pass
+    = retryVerification 0 $
+        if shouldCheckLocType mode rInfo
+        then case rInfo of
+          RIFileLocal -> checkRef rAnchor (riRoot repoInfo) file ""
+          RIFileRelative ->
+            let relativeToRoot = getRelativeOrAbsoluteChild (riRoot repoInfo) (takeDirectory file)
+                  </> toString rLink
+             in if hasIndirectionThroughParent relativeToRoot
```

Comment:
These checks make links that pass through root considered as being outside the repository. Is this what you requested @Martoon-00?

Reply:
I wanted this to be checked upon `CanonicalPath` construction, but apparently this is now not justified, my previous concern about poking files outside of the repository does not hold for `makeAbsolute`. So nevermind my previous comment.

This looks good, however compared to how this check was performed previously, we lost the behaviour that this check is not performed if the file is considered virtual. I suspect we have to preserve this behaviour.